cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/AbstractCompositeTrigger.java

```diff
@@ -111,4 +130,15 @@ private void initializeUnitTriggers() {
     }
     return unitTriggerInfos.build();
   }
+
+  /**
+   * Return the simplified version of the current composite trigger based on the condition that the given program
+   * is deleted so that the {@link co.cask.cdap.internal.app.runtime.schedule.trigger.ProgramStatusTrigger} of
+   * the given program contained in the current composite trigger will never be satisfied.
+   *
+   * @param programId the program id of the deleted program
+   * @return the simplified composite trigger, or {@code null} if the composite trigger will never be satisfied
+   */
+  @Nullable
+  public abstract AbstractCompositeTrigger getTriggerWithDeletedProgram(ProgramId programId);
```

Comment:
this won't always be composite right? If you have program1 or program2 and remove program2, don't you end up with a ProgramStatusTrigger?